src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingCook.java

```diff
@@ -520,8 +520,8 @@ public IRecipeStorage getFirstRecipe(final Predicate<ItemStack> stackPredicate)
 
             //If we didn't have a stored recipe, see if there is a smelting recipe that is also a food output, and use it.
             storage = FurnaceRecipes.getInstance().getFirstSmeltingRecipeByResult(stackPredicate);
-            Optional<Boolean> validRecipe =  canRecipeBeAddedBasedOnTags(storage);
-            if(storage != null && ISFOOD.test(storage.getPrimaryOutput().getStack()) && (!validRecipe.isPresent() || validRecipe.get()))
+            boolean validRecipe = isRecipeCompatibleWithCraftingModule(storage.getToken());
```

Comment:
`isRecipeCompatibleWithCraftingModule(storage.getToken())` => `isRecipeCompatible(GenericRecipe.of(storage))`

A bit more efficient, since it's going "forwards" in the data chain and doesn't have to look up the recipe again.  _Technically_ the latter is nullable but it only actually returns null if the parameter is null, so this usage is safe.

Reply:
(Also probably `isRecipeCompatibleWithCraftingModule` should just become another overload of `isRecipeCompatible` since it's actually **on** the crafting module now, instead of on the building.  But that's optional.)